src/software/ai/motion_constraint/motion_constraint_set_builder.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <set>
+
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/hl/stp/tactic/tactic_visitor.h"
+#include "software/ai/motion_constraint/motion_constraint.h"
+#include "software/world/game_state.h"
+
+
+/**
+ * Builds motion constraints based on gamestate and tactic
+ *
+ * @param gamestate Current GameState to process
+ * @param tactic Current Tactic to process
+ *
+ * @return set of MotionConstraints
+ */
+std::set<MotionConstraint> buildMotionConstraintSet(const GameState &game_state,
+                                                    Tactic &tactic);
```

Comment:
can we have `const Tactic&`? (this might have changed from https://github.com/UBC-Thunderbots/Software/commit/7061df8fe31e82f699a4a4624dd1efa1a49ee914, so please merge in `upstream/master`)